homeassistant/components/backup/config.py

```diff
@@ -176,12 +191,36 @@ def update(
         """Update config."""
         if agents is not UNDEFINED:
             for agent_id, agent_config in agents.items():
-                if agent_id not in self.data.agents:
-                    self.data.agents[agent_id] = AgentConfig(**agent_config)
+                agent_retention = agent_config.get("retention", UNDEFINED)
```

Comment:
Why do we use UNDEFINED instead of None?

Reply:
Although, I may have missed to finish that thought in the code. I'll double check.

Reply:
I don't think we need this since we use another membership check below.

Reply:
```suggestion
                agent_retention = agent_config.get("retention")
```